STM32Cube/Core/Inc/FreeRTOSConfig.h

```diff
@@ -68,7 +68,7 @@
 #define configTICK_RATE_HZ                       ((TickType_t)1000)
 #define configMAX_PRIORITIES                     ( 56 )
 #define configMINIMAL_STACK_SIZE                 ((uint16_t)128)
-#define configTOTAL_HEAP_SIZE                    ((size_t)15360)
+#define configTOTAL_HEAP_SIZE                    ((size_t)16360)
```

Comment:
I think we can make this way higher. This is where FreeRTOS pulls from when you declare an RTOS object (task, queue, semaphore, etc). Which means for example for all tasks to get 2000 words of stack space, you need at least 2000 * #tasks * 2 bytes per word, just for task space. So like, 50k bytes isn't out there. Reminder that our chip has 512 kB of RAM so that's not at all unreasonable

Reply:
i will be looking into how much heap is needed, but yeah always start with overallocating.